compiler/src/dotty/tools/dotc/core/Flags.scala

```diff
@@ -172,12 +172,24 @@ object Flags {
   }
 
   /** The union of all flags in given flag set */
-  def union(flagss: FlagSet*) = (EmptyFlags /: flagss)(_ | _)
+  def union(flagss: FlagSet*): FlagSet = {
+    var flag = EmptyFlags
+    for (f <- flagss)
```

Comment:
If you want to go deep in micro-optimizations, you could use a while loop

Reply:
Should avoid `FlagSet*` altogether.